pkg/signal/signal_test.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package signal
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"storj.io/storj/pkg/utils"
+)
+
+func TestDispatcher(t *testing.T) {
+	d := NewDispatcher("test")
+	assert.Equal(t, d.source, "test")
+	assert.NotNil(t, d)
+}
+
+func TestRegistrationAndDispatch(t *testing.T) {
+	d := NewDispatcher("test")
+	d.Register("test:error", func() error {
+		return errors.New("error")
+	})
+
+	err := d.Dispatch("test:error")
+	assert.Equal(t, err, errors.New("error"))
+
+	err = d.Register("test:normal", func() error {
+		return nil
+	})
+	err = d.Dispatch("test:normal")
+	assert.Equal(t, err, nil)
+
+	err = d.Register("test:multiple", func() error {
+		return nil
```

Comment:
do you want any of your tests to do something in the callback, e.g., increment a counter or something, so you know it did stuffs?